lib/vsc/utils/generaloption.py

```diff
@@ -106,6 +107,25 @@ def check_str_list_tuple(option, opt, value):
         return klass(split)
 
 
+def check_dict(option, opt, value):
+    """
+    Check function for dict type
```

Comment:
can we have a valid use case for such a horrible self invented issue? why not support JSON, at least it's well defined.